.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,64 @@
+name: CI Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  ci-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version:
+        - python: "2.7"
+          tornado: ">=4,<5"
+        - python: "2.7"
+          tornado: ">=5,<6"
+        - python: "3.5"
+          tornado: ">=4,<5"
+        - python: "3.5"
+          tornado: ">=5,<6"
+        - python: "3.5"
+          tornado: ">=6"
+        - python: "3.6"
+          tornado: ">=4,<5"
+        - python: "3.6"
+          tornado: ">=5,<6"
+        - python: "3.6"
+          tornado: ">=6"
+        - python: "3.7"
+          tornado: ">=4,<5"
+        - python: "3.7"
+          tornado: ">=5,<6"
+        - python: "3.7"
+          tornado: ">=6"
+    name: ci-tests (py ${{ matrix.version.python }} tornado ${{ matrix.version.tornado }})
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.version.python }}
+
+    - name: Export TORNADO variable
+      run: |
+        echo "TORNADO=${{ matrix.version.tornado }}" >> $GITHUB_ENV
+
+    - name: Install pycurl dependencies
+      run: sudo apt-get install libgnutls28-dev libcurl4-openssl-dev libssl-dev
+
+    - name: Install dependencies
+      run: make bootstrap
+
+    - name: Run tests
+      run: make test_ci
+
+    - name: Upload coverage to codecov
+      uses: codecov/codecov-action@v1
```

Comment:
Does this action support accumulation of coverage from multiple runs? I think there are some code branches that are sensitive to the versions matrix, and we compute coverage in every run anyway, it would make sense to aggregate.

Reply:
From https://docs.codecov.io/docs/merging-reports 
> Automatic report merging for all CI and languages into a single report.

Seems like codecov already does that.

Reply:
Yes, but the example in that link only calls codecov once, to submit multiple results in one bacth. In our case each job will call it, and I am not sure how codecov will know that those are all for the same PR, unless the action itself is smart enough to recognize that.

Reply:
Not sure how the merged report should look like, but I can see the 11 uploads from the 11 CI jobs - https://codecov.io/gh/Ashmita152/jaeger-client-python/commit/092d2fe6a6f625924178bc503bbb2beac7aa0b5b/build 

And looking at the repo for pyra/cryptography (which they have mentioned in the above doc) https://codecov.io/gh/pyca/cryptography/commit/7d60786a72ed09fc53b5f7bd5e2c89166dccbbfa/build

Both look similar to me. 

From above doc: 

> Codecov does not override report data for multiple uploads. We always merge the data. Simply upload all three reports at once, or separately.

This feels like we don't need to do anything special on our side and just upload.